nexusmods-app: use `enableUnfree` in name + description

MattSturgeon opened the pull request:
## Description of changes

- Append `-unfree` to package name
- Append whether or not RAR format mods are supported to the description
- Format files using nixfmt
	- Do this automatically in the update script too
	- Enforce this in the check-formatting CI workflow
- Simplify some string concatenation
- Refactor how the tests are declared (use `mapAttrs`)

`nexusmods-app`'s description:
> Game mod installer, creator and manager. Use `nexusmods-app-unfree` instead if you need support for mods packaged using the unfree RAR format.

`nexusmods-app-unfree` description:
> Game mod installer, creator and manager. Includes support for mods packaged using the unfree RAR format.


<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

MattSturgeon commented:
I considered adding the package to the check-nixfmt workflow, however that'd mean we'd have to format deps.nix too.

<details><summary>Details</summary>
<p>

```diff
diff --git a/.github/workflows/check-nix-format.yml b/.github/workflows/check-nix-format.yml
index 131803213cb5..999955e352b9 100644
--- a/.github/workflows/check-nix-format.yml
+++ b/.github/workflows/check-nix-format.yml
@@ -32,6 +32,8 @@ jobs:
         # Each environment variable beginning with NIX_FMT_PATHS_ is a list of
         # paths to check with nixfmt.
         env:
+          NIX_FMT_PATHS_BY_NAME: |
+            pkgs/by-name/ne/nexusmods-app
           NIX_FMT_PATHS_BSD: pkgs/os-specific/bsd
           NIX_FMT_PATHS_MPVSCRIPTS: pkgs/applications/video/mpv/scripts
           # Format paths related to the Nixpkgs CUDA ecosystem.
```

</p>
</details>

l0b0 commented:
> I considered adding the package to the [check-nixfmt workflow](https://github.com/NixOS/nixpkgs/blob/master/.github/workflows/check-nix-format.yml?rgh-link-date=2024-06-21T03%3A33%3A18Z), however that'd mean we'd have to format deps.nix too.

I'd be in favour of adding it, even if we have to reformat deps.nix. Making all the code (including generated code) easier to read will help.

l0b0 reviewed (approved):
LGTM, thanks!

MattSturgeon committed:
nexusmods-app: format with nixfmt

- reformatted using nixfmt (RFC166)
- added nixfmt to the update script
- simplify some string concatenation
- refactor tests declaration

MattSturgeon committed:
github/workflows: add nexusmods-app to check-nix-format

MattSturgeon committed:
nexusmods-app: use `enableUnfree` in name + description

- Append `-unfree` to package name
- Append whether or not RAR format mods are supported to the description

MattSturgeon commented:
> > I considered adding the package to the [check-nixfmt workflow](https://github.com/NixOS/nixpkgs/blob/master/.github/workflows/check-nix-format.yml?rgh-link-date=2024-06-21T04%3A05%3A47Z), however that'd mean we'd have to format deps.nix too.
> 
> I'd be in favour of adding it, even if we have to reformat deps.nix. Making all the code (including generated code) easier to read will help.

Ok, I've updated the PR to do that then:

- First commit reformats, cleans up some code, and adds nixfmt to the update script
- Second commit adds nexusmods-app to the check-format CI workflow
- Third commit is the main focus of the PR; adding unfree-specific name & description

MattSturgeon commented:
I don't see us in the check-format logs, but I guess that's cos the security policy enforces using the workflow file from master?

<details><summary>Logs</summary>
<p>

```
Checking paths: pkgs/os-specific/bsd
Checking paths: pkgs/build-support/php
Checking paths: pkgs/development/cuda-modules pkgs/test/cuda pkgs/top-level/cuda-packages.nix 
Checking paths: nixos/modules/services/cluster/k3s nixos/tests/k3s pkgs/applications/networking/cluster/k3s 
Checking paths: maintainers/maintainer-list.nix maintainers/team-list.nix 
Checking paths: pkgs/applications/video/mpv/scripts
Checking paths: pkgs/development/php-packages
Checking paths: pkgs/applications/editors/vscode/extensions
```
https://github.com/NixOS/nixpkgs/actions/runs/9618191097/job/26531550883?pr=321405

</p>
</details>

MattSturgeon commented:
Updated the new description to comply with https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

In particular:
- Be short, just one sentence.
- More generally, it should not refer to the package name.
- Not end with a period (or any punctuation for that matter).

l0b0 commented:
> I'm tempted to simplifying that part down to just _"Game mod manager"_, since we probably don't need to be as verbose as upstream here.

Agreed. We absolutely don't need to copy upstream, that was just the most convenient place to start.

l0b0 commented on pkgs/by-name/ne/nexusmods-app/package.nix:
```diff
@@ -63,52 +60,53 @@ buildDotnetModule rec {
-        touch $out
-      '';
-    };
+    tests =
+      lib.attrsets.mapAttrs
+        (
+          tname: args:
+          runCommand "${pname}-test-${tname}" { } ''
+            ${lib.getExe nexusmods-app} ${args}
+            touch $out
+          ''
+        )
+        {
+          serve = "";
+          help = "--help";
+          associate-nxm = "associate-nxm";
+          list-tools = "list-tools";
```
Looking at this again, it feels a bit like premature abstraction. But I'm pretty new to nixpkgs, so I'd be interested to hear what others think about it. Don't feel like you have to change it back; I'm still on the fence about whether the complexity is worth the de-duplication.

MattSturgeon replied:
Interesting, I actually felt like this abstraction _simplified_ things rather than adding complexity. But it's not a necessary change.

l0b0 replied:
Nah, feel free to leave it in place then.

l0b0 commented on pkgs/by-name/ne/nexusmods-app/package.nix:
```diff
@@ -63,52 +60,53 @@ buildDotnetModule rec {
+        (
+          tname: args:
+          runCommand "${pname}-test-${tname}" { } ''
+            ${lib.getExe nexusmods-app} ${args}
```
I like the simpler way to get the executable. :+1:

MattSturgeon commented on .github/workflows/check-nix-format.yml:
```diff
@@ -32,6 +32,8 @@ jobs:
         # Each environment variable beginning with NIX_FMT_PATHS_ is a list of
         # paths to check with nixfmt.
         env:
+          NIX_FMT_PATHS_BY_NAME: |
+            pkgs/by-name/ne/nexusmods-app
```
It'll reduce future diff sizes if other by-name packages are added, consider these diffs:

```diff
@@ -32,7 +32,9 @@ jobs:
         # Each environment variable beginning with NIX_FMT_PATHS_ is a list of
         # paths to check with nixfmt.
         env:
-          NIX_FMT_PATHS_BY_NAME: pkgs/by-name/ne/nexusmods-app
+          NIX_FMT_PATHS_BY_NAME: |
+            pkgs/by-name/ne/nexusmods-app
+            pkgs/by-name/so/some-other
           NIX_FMT_PATHS_BSD: pkgs/os-specific/bsd
           NIX_FMT_PATHS_MPVSCRIPTS: pkgs/applications/video/mpv/scripts
           # Format paths related to the Nixpkgs CUDA ecosystem.
```
vs
```diff
--- a/.github/workflows/check-nix-format.yml
+++ b/.github/workflows/check-nix-format.yml
@@ -34,6 +34,7 @@ jobs:
         env:
           NIX_FMT_PATHS_BY_NAME: |
             pkgs/by-name/ne/nexusmods-app
+            pkgs/by-name/so/some-other
           NIX_FMT_PATHS_BSD: pkgs/os-specific/bsd
           NIX_FMT_PATHS_MPVSCRIPTS: pkgs/applications/video/mpv/scripts
           # Format paths related to the Nixpkgs CUDA ecosystem.
```

l0b0 replied:
:thinking: In general I'd probably take the standard automated formatter approach and take that hit *once* if we ever actually extend the list. But again, just nitpicking.

MattSturgeon replied:
I'm not aware of an automated formatter being used for nixpkgs yaml files, but I'll format it however the ci/workflow maintainers prefer.

MattSturgeon commented:
> > I'm tempted to simplifying that part down to just _"Game mod manager"_, since we probably don't need to be as verbose as upstream here.
> 
> Agreed. We absolutely don't need to copy upstream, that was just the most convenient place to start.

Done

Mic92 commented on pkgs/by-name/ne/nexusmods-app/update.bash:
```diff
@@ -1,9 +1,10 @@
 set -o errexit -o nounset -o pipefail
 
 package=nexusmods-app
 nix-update "$package"
 "$(nix-build --attr "$package".fetch-deps --no-out-link)"
```
Interesting. Is this something that nix-update maybe should implement?

MattSturgeon replied:
Perhaps. I think it's used commonly throughout dotnet packages, IDK how widespread it is elsewhere.

Mic92 commented:
Result of `nixpkgs-review pr 321405` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>nexusmods-app</li>
    <li>nexusmods-app-unfree</li>
  </ul>
</details>

Mic92 commented:
I get the following error:
  
```
$ nix log /nix/store/v8r92xhslchilr1mr8jmhhmq4a6h0y7k-nexusmods-app-0.4.1.drv 
...
Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
[xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.5.7+8f2703126a (64-bit .NET 8.0.6)
[xUnit.net 00:00:00.08]   Discovering: NexusMods.Abstractions.DataModel.Entities.Tests
[xUnit.net 00:00:00.10]   Discovered:  NexusMods.Abstractions.DataModel.Entities.Tests
[xUnit.net 00:00:00.11]   Starting:    NexusMods.Abstractions.DataModel.Entities.Tests
[xUnit.net 00:00:00.80]   Finished:    NexusMods.Abstractions.DataModel.Entities.Tests
  Passed NexusMods.Abstractions.DataModel.Entities.Tests.ModelTests.CanCreateModFile [232 ms]

Test Run Successful.
Total tests: 1
     Passed: 1
 Total time: 1.3058 Seconds
Test run for /build/source/tests/Abstractions/NexusMods.Abstractions.Games.Tests/bin/Release/net8.0/NexusMods.Abstractions.Games.Tests.dll (.NETCoreApp,Version=v8.0)
Microsoft (R) Test Execution Command Line Tool Version 17.10.0 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
[xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.5.7+8f2703126a (64-bit .NET 8.0.6)
[xUnit.net 00:00:00.09]   Discovering: NexusMods.Abstractions.Games.Tests
[xUnit.net 00:00:00.15]   Discovered:  NexusMods.Abstractions.Games.Tests
[xUnit.net 00:00:00.16]   Starting:    NexusMods.Abstractions.Games.Tests
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetResolvedPathTest(id: "AppData", relativePath: "", expected: "/foo/qux/AppData/Local/Skyrim") [36 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetResolvedPathTest(id: "Saves", relativePath: "foo/bar", expected: "/foo/baz/documents/My Games/Skyrim/Saves/foo/bar") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetResolvedPathTest(id: "Documents", relativePath: "", expected: "/foo/baz/documents/My Games/Skyrim") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetResolvedPathTest(id: "Game", relativePath: "foo", expected: "/foo/bar/Skyrim/foo") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetResolvedPathTest(id: "Preferences", relativePath: "foo/bar", expected: "/foo/baz/documents/My Games/Skyrim/foo/bar") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetResolvedPathTest(id: "SKSE Plugins", relativePath: "", expected: "/foo/bar/Skyrim/data/skse/plugins") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetResolvedPathTest(id: "Game", relativePath: "", expected: "/foo/bar/Skyrim") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetResolvedPathTest(id: "Saves", relativePath: "", expected: "/foo/baz/documents/My Games/Skyrim/Saves") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetResolvedPathTest(id: "Documents", relativePath: "foo/bar", expected: "/foo/baz/documents/My Games/Skyrim/foo/bar") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetResolvedPathTest(id: "Data", relativePath: "foo/bar", expected: "/foo/bar/Skyrim/data/foo/bar") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetResolvedPathTest(id: "Data", relativePath: "", expected: "/foo/bar/Skyrim/data") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetResolvedPathTest(id: "SKSE Plugins", relativePath: "foo/bar/qux", expected: "/foo/bar/Skyrim/data/skse/plugins/foo/bar/qux") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetResolvedPathTest(id: "AppData", relativePath: "foo/bar", expected: "/foo/qux/AppData/Local/Skyrim/foo/bar") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetResolvedPathTest(id: "Preferences", relativePath: "", expected: "/foo/baz/documents/My Games/Skyrim") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.ToGamePathTest(absolutePath: "/foo/baz/documents/My Games/Skyrim/Saves", expectedId: "Saves", expectedRelativePath: "") [5 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.ToGamePathTest(absolutePath: "/foo/bar/Skyrim/data/foo/bar", expectedId: "Data", expectedRelativePath: "foo/bar") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.ToGamePathTest(absolutePath: "/foo/bar/Skyrim", expectedId: "Game", expectedRelativePath: "") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.ToGamePathTest(absolutePath: "/foo/baz/documents/My Games/Skyrim/foo/bar", expectedId: "Preferences", expectedRelativePath: "foo/bar") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.ToGamePathTest(absolutePath: "/foo/bar/Skyrim/data", expectedId: "Data", expectedRelativePath: "") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.ToGamePathTest(absolutePath: "/foo/qux/AppData/Local/Skyrim", expectedId: "AppData", expectedRelativePath: "") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.ToGamePathTest(absolutePath: "/foo/bar/Skyrim/foo", expectedId: "Game", expectedRelativePath: "foo") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.ToGamePathTest(absolutePath: "/foo/baz/documents/My Games/Skyrim", expectedId: "Preferences", expectedRelativePath: "") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.ToGamePathTest(absolutePath: "/foo/qux/AppData/Local/Skyrim/foo/bar", expectedId: "AppData", expectedRelativePath: "foo/bar") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.ToGamePathTest(absolutePath: "/foo/baz/documents/My Games/Skyrim/Saves/foo/bar", expectedId: "Saves", expectedRelativePath: "foo/bar") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.ToGamePathTest(absolutePath: "/foo/bar/Skyrim/data/skse/plugins", expectedId: "SKSE Plugins", expectedRelativePath: "") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.ToGamePathTest(absolutePath: "/foo/bar/Skyrim/data/skse/plugins/foo/bar/qux", expectedId: "SKSE Plugins", expectedRelativePath: "foo/bar/qux") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.IsTopLevelTest(id: "Game", expected: True) [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.IsTopLevelTest(id: "Saves", expected: False) [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.IsTopLevelTest(id: "Documents", expected: True) [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.IsTopLevelTest(id: "Preferences", expected: True) [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.IsTopLevelTest(id: "AppData", expected: True) [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.IsTopLevelTest(id: "SKSE Plugins", expected: False) [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.IsTopLevelTest(id: "Data", expected: False) [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.IndexTest(id: "Preferences", expected: "/foo/baz/documents/My Games/Skyrim", found: True) [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.IndexTest(id: "Unknown", expected: "", found: False) [7 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.IndexTest(id: "SKSE Plugins", expected: "/foo/bar/Skyrim/data/skse/plugins", found: True) [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.IndexTest(id: "Saves", expected: "/foo/baz/documents/My Games/Skyrim/Saves", found: True) [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.IndexTest(id: "Data", expected: "/foo/bar/Skyrim/data", found: True) [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.IndexTest(id: "Game", expected: "/foo/bar/Skyrim", found: True) [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.IndexTest(id: "Documents", expected: "/foo/baz/documents/My Games/Skyrim", found: True) [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.IndexTest(id: "game", expected: "", found: False) [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.IndexTest(id: "AppData", expected: "/foo/qux/AppData/Local/Skyrim", found: True) [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetNestedLocationsTest(id: "SKSE Plugins", expectedChildren: []) [29 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetNestedLocationsTest(id: "Game", expectedChildren: ["Data", "SKSE Plugins"]) [6 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetNestedLocationsTest(id: "Saves", expectedChildren: []) [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetNestedLocationsTest(id: "Documents", expectedChildren: ["Saves"]) [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetNestedLocationsTest(id: "Preferences", expectedChildren: ["Saves"]) [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetNestedLocationsTest(id: "AppData", expectedChildren: []) [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetNestedLocationsTest(id: "Data", expectedChildren: ["SKSE Plugins"]) [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetTopLevelParentTest(id: "Preferences", expected: "Preferences") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetTopLevelParentTest(id: "AppData", expected: "AppData") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetTopLevelParentTest(id: "Documents", expected: "Documents") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetTopLevelParentTest(id: "Data", expected: "Game") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetTopLevelParentTest(id: "SKSE Plugins", expected: "Game") [< 1 ms]
[xUnit.net 00:00:00.49]   Finished:    NexusMods.Abstractions.Games.Tests
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetTopLevelParentTest(id: "Game", expected: "Game") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetTopLevelParentTest(id: "Saves", expected: "Preferences") [< 1 ms]
  Passed NexusMods.Abstractions.Games.Tests.GameLocationsRegisterTests.GetTopLevelLocationsTest [31 ms]

Test Run Successful.
Total tests: 57
     Passed: 57
 Total time: 1.1012 Seconds
Test run for /build/source/tests/NexusMods.App.Generators.Diagnostics.Tests/bin/Release/net8.0/NexusMods.App.Generators.Diagnostics.Tests.dll (.NETCoreApp,Version=v8.0)
Microsoft (R) Test Execution Command Line Tool Version 17.10.0 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
[xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.5.7+8f2703126a (64-bit .NET 8.0.6)
[xUnit.net 00:00:00.07]   Discovering: NexusMods.App.Generators.Diagnostics.Tests
[xUnit.net 00:00:00.11]   Discovered:  NexusMods.App.Generators.Diagnostics.Tests
[xUnit.net 00:00:00.11]   Starting:    NexusMods.App.Generators.Diagnostics.Tests
[xUnit.net 00:00:01.11]   Finished:    NexusMods.App.Generators.Diagnostics.Tests
  Passed NexusMods.App.Generators.Diagnostics.Tests.DiagnosticTemplateIncrementalSourceGeneratorTests.TestGenerator [932 ms]

Test Run Successful.
Total tests: 1
     Passed: 1
 Total time: 1.6317 Seconds
Finished dotnetCheckHook
checkPhase completed in 1 minutes 27 seconds
@nix {"action":"setPhase","phase":"installPhase"}
Running phase: installPhase
Executing dotnetInstallHook
/nix/store/4mmpxd5ifbin8dxi2pd8hrsgbycr097a-dotnet-sdk-8.0.301/sdk/8.0.301/Current/SolutionFile/ImportAfter/Microsoft.NET.Sdk.Solution.targets(36,5): warning NETSDK1194: The "--output" option isn't supported when building a solution. Specifying a solution-level output path results in all projects copying outputs to the same directory, which can lead to inconsistent builds. [/build/source/NexusMods.App.sln]
  NexusMods.Abstractions.Cli -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Extensions.DependencyInjection -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Abstractions.Messaging -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Abstractions.IO -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Abstractions.Settings -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Abstractions.Activities -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Extensions.BCL -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.App.BuildInfo -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.CrossPlatform -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Abstractions.Triggers -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Abstractions.GameLocators -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Icons -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.StardewValley.SMAPI -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.App.Generators.Diagnostics -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Settings -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Abstractions.Serialization -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
/nix/store/4mmpxd5ifbin8dxi2pd8hrsgbycr097a-dotnet-sdk-8.0.301/sdk/8.0.301/Microsoft.Common.CurrentVersion.targets(2165,5): warning : The referenced project '../Abstractions/NexusMods.Abstractions.App.Settings/NexusMods.Abstractions.App.Settings.csproj' does not exist. [/build/source/src/NexusMods.DataModel/NexusMods.DataModel.csproj]
  NexusMods.Extensions.DynamicData -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Abstractions.HttpDownloader -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/Microsoft.CodeCoverage.targets(23,5): warning MSB3026: Could not copy "/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/InstrumentationEngine/x86/MicrosoftInstrumentationEngine_x86.dll" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/InstrumentationEngine/x86/MicrosoftInstrumentationEngine_x86.dll". Beginning retry 1 in 1000ms. The process cannot access the file '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/InstrumentationEngine/x86/MicrosoftInstrumentationEngine_x86.dll' because it is being used by another process.  [/build/source/tests/Abstractions/NexusMods.Abstractions.GuidedInstallers.Tests/NexusMods.Abstractions.GuidedInstallers.Tests.csproj]
  NexusMods.Abstractions.GuidedInstallers -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Abstractions.FileExtractor -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Extensions.Hashing -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Abstractions.DiskState -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Abstractions.NexusWebApi -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Abstractions.Installers -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Networking.HttpDownloader -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Activities -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
/build/tmp.dKYu2Rpy6p/.nuget/packages/coverlet.collector/6.0.0/build/netstandard1.0/coverlet.collector.targets(17,5): warning MSB3026: Could not copy "/build/tmp.dKYu2Rpy6p/.nuget/packages/coverlet.collector/6.0.0/build/netstandard1.0/Mono.Cecil.dll" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/Mono.Cecil.dll". Beginning retry 1 in 1000ms. The process cannot access the file '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/Mono.Cecil.dll' because it is being used by another process.  [/build/source/tests/Abstractions/NexusMods.Abstractions.Installers.Tests/NexusMods.Abstractions.Installers.Tests.csproj]
  NexusMods.Abstractions.FileStore -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.FileExtractor -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Abstractions.Loadouts -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.BCL.Extensions.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.AdvancedInstaller -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.FOMOD -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Abstractions.Serialization.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Abstractions.Loadouts.Synchronizers -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/Microsoft.CodeCoverage.targets(23,5): warning MSB3026: Could not copy "/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/InstrumentationEngine/x64/MicrosoftInstrumentationEngine_x64.dll" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/InstrumentationEngine/x64/MicrosoftInstrumentationEngine_x64.dll". Beginning retry 1 in 1000ms. The process cannot access the file '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/InstrumentationEngine/x64/MicrosoftInstrumentationEngine_x64.dll' because it is being used by another process.  [/build/source/tests/NexusMods.CrossPlatform.Tests/NexusMods.CrossPlatform.Tests.csproj]
/nix/store/4mmpxd5ifbin8dxi2pd8hrsgbycr097a-dotnet-sdk-8.0.301/sdk/8.0.301/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Publish.targets(351,5): warning MSB3026: Could not copy "/build/source/src/Abstractions/NexusMods.Abstractions.Games/bin/Release/net8.0/NexusMods.Abstractions.Games.pdb" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/NexusMods.Abstractions.Games.pdb". Beginning retry 1 in 1000ms. The process cannot access the file '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/NexusMods.Abstractions.Games.pdb' because it is being used by another process.  [/build/source/src/Abstractions/NexusMods.Abstractions.Games.Diagnostics/NexusMods.Abstractions.Games.Diagnostics.csproj]
/nix/store/4mmpxd5ifbin8dxi2pd8hrsgbycr097a-dotnet-sdk-8.0.301/sdk/8.0.301/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Publish.targets(351,5): warning MSB3026: Could not copy "/build/source/src/Abstractions/NexusMods.Abstractions.Games/bin/Release/net8.0/NexusMods.Abstractions.Games.xml" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/NexusMods.Abstractions.Games.xml". Beginning retry 1 in 1000ms. The process cannot access the file '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/NexusMods.Abstractions.Games.xml' because it is being used by another process.  [/build/source/src/Networking/NexusMods.Networking.Downloaders/NexusMods.Networking.Downloaders.csproj]
  NexusMods.Abstractions.Games -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
/nix/store/4mmpxd5ifbin8dxi2pd8hrsgbycr097a-dotnet-sdk-8.0.301/sdk/8.0.301/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Publish.targets(351,5): warning MSB3026: Could not copy "/build/source/src/Abstractions/NexusMods.Abstractions.Games/bin/Release/net8.0/NexusMods.Abstractions.Games.pdb" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/NexusMods.Abstractions.Games.pdb". Beginning retry 1 in 1000ms. The process cannot access the file '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/NexusMods.Abstractions.Games.pdb' because it is being used by another process.  [/build/source/src/Games/NexusMods.Games.DarkestDungeon/NexusMods.Games.DarkestDungeon.csproj]
  NexusMods.Games.BladeAndSorcery -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.App.Cli -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.Generic -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Networking.HttpDownloader.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Networking.NexusWebApi -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.Reshade -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.TestHarness -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.RedEngine -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.Sifu -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.MountAndBlade2Bannerlord -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.BethesdaGameStudios -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.StandardGameLocators -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.App.Generators.Diagnostics.Sample -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.StardewValley -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Abstractions.Games.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
/nix/store/4mmpxd5ifbin8dxi2pd8hrsgbycr097a-dotnet-sdk-8.0.301/sdk/8.0.301/Microsoft.Common.CurrentVersion.targets(2165,5): warning : The referenced project '../Abstractions/NexusMods.Abstractions.App.Settings/NexusMods.Abstractions.App.Settings.csproj' does not exist. [/build/source/src/NexusMods.App.UI/NexusMods.App.UI.csproj]
  NexusMods.App.Generators.Diagnostics.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.StandardGameLocators.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
/build/tmp.dKYu2Rpy6p/.nuget/packages/coverlet.collector/6.0.0/build/netstandard1.0/coverlet.collector.targets(17,5): warning MSB3026: Could not copy "/build/tmp.dKYu2Rpy6p/.nuget/packages/coverlet.collector/6.0.0/build/netstandard1.0/Mono.Cecil.Pdb.dll" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/Mono.Cecil.Pdb.dll". Beginning retry 1 in 1000ms. The process cannot access the file '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/Mono.Cecil.Pdb.dll' because it is being used by another process.  [/build/source/tests/Games/NexusMods.Games.TestFramework/NexusMods.Games.TestFramework.csproj]
/build/tmp.dKYu2Rpy6p/.nuget/packages/coverlet.collector/6.0.0/build/netstandard1.0/coverlet.collector.targets(17,5): warning MSB3026: Could not copy "/build/tmp.dKYu2Rpy6p/.nuget/packages/coverlet.collector/6.0.0/build/netstandard1.0/Mono.Cecil.dll" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/Mono.Cecil.dll". Beginning retry 1 in 1000ms. The process cannot access the file '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/Mono.Cecil.dll' because it is being used by another process.  [/build/source/tests/Games/NexusMods.Games.TestFramework/NexusMods.Games.TestFramework.csproj]
  NexusMods.Benchmarks -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.FileExtractor.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/Microsoft.CodeCoverage.targets(23,5): warning MSB3026: Could not copy "/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/CodeCoverage/arm64/covrunarm64.dll" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/CodeCoverage/arm64/covrunarm64.dll". Beginning retry 1 in 1000ms. The process cannot access the file '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/CodeCoverage/arm64/covrunarm64.dll' because it is being used by another process.  [/build/source/tests/Games/NexusMods.Games.RedEngine.Tests/NexusMods.Games.RedEngine.Tests.csproj]
/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/Microsoft.CodeCoverage.targets(23,5): warning MSB3026: Could not copy "/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/CodeCoverage/arm64/VanguardInstrumentationProfiler_arm64.config" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/CodeCoverage/arm64/VanguardInstrumentationProfiler_arm64.config". Beginning retry 1 in 1000ms. The process cannot access the file '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/CodeCoverage/arm64/VanguardInstrumentationProfiler_arm64.config' because it is being used by another process.  [/build/source/tests/Games/NexusMods.Games.StardewValley.Tests/NexusMods.Games.StardewValley.Tests.csproj]
/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/Microsoft.CodeCoverage.targets(23,5): warning MSB3026: Could not copy "/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/CodeCoverage/arm64/msdia140.dll" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/CodeCoverage/arm64/msdia140.dll". Beginning retry 1 in 1000ms. The process cannot access the file '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/CodeCoverage/arm64/msdia140.dll' because it is being used by another process.  [/build/source/tests/Games/NexusMods.Games.StardewValley.Tests/NexusMods.Games.StardewValley.Tests.csproj]
/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/Microsoft.CodeCoverage.targets(23,5): warning MSB3026: Could not copy "/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/CodeCoverage/amd64/CodeCoverage.exe" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/CodeCoverage/amd64/CodeCoverage.exe". Beginning retry 1 in 1000ms. The process cannot access the file '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/CodeCoverage/amd64/CodeCoverage.exe' because it is being used by another process.  [/build/source/tests/Games/NexusMods.Games.AdvancedInstaller.Tests/NexusMods.Games.AdvancedInstaller.Tests.csproj]
  NexusMods.Abstractions.DataModel.Entities.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/Microsoft.CodeCoverage.targets(23,5): warning MSB3026: Could not copy "/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/CodeCoverage/msdia140.dll" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/CodeCoverage/msdia140.dll". Beginning retry 1 in 1000ms. The process cannot access the file '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/CodeCoverage/msdia140.dll' because it is being used by another process.  [/build/source/tests/Games/NexusMods.Games.MountAndBlade2Bannerlord.Tests/NexusMods.Games.MountAndBlade2Bannerlord.Tests.csproj]
/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/Microsoft.CodeCoverage.targets(23,5): warning MSB3026: Could not copy "/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/CodeCoverage/VanguardInstrumentationProfiler_x86.config" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/CodeCoverage/VanguardInstrumentationProfiler_x86.config". Beginning retry 1 in 1000ms. The process cannot access the file '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/CodeCoverage/VanguardInstrumentationProfiler_x86.config' because it is being used by another process.  [/build/source/tests/Games/NexusMods.Games.MountAndBlade2Bannerlord.Tests/NexusMods.Games.MountAndBlade2Bannerlord.Tests.csproj]
/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/Microsoft.CodeCoverage.targets(23,5): warning MSB3026: Could not copy "/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/CodeCoverage/CodeCoverage.exe" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/CodeCoverage/CodeCoverage.exe". Beginning retry 1 in 1000ms. The process cannot access the file '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/CodeCoverage/CodeCoverage.exe' because it is being used by another process.  [/build/source/tests/Games/NexusMods.Games.MountAndBlade2Bannerlord.Tests/NexusMods.Games.MountAndBlade2Bannerlord.Tests.csproj]
/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/Microsoft.CodeCoverage.targets(23,5): warning MSB3026: Could not copy "/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/CodeCoverage/codecoveragemessages.dll" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/CodeCoverage/codecoveragemessages.dll". Beginning retry 1 in 1000ms. The process cannot access the file '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/CodeCoverage/codecoveragemessages.dll' because it is being used by another process.  [/build/source/tests/Games/NexusMods.Games.MountAndBlade2Bannerlord.Tests/NexusMods.Games.MountAndBlade2Bannerlord.Tests.csproj]
/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/Microsoft.CodeCoverage.targets(23,5): warning MSB3026: Could not copy "/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/CodeCoverage/covrun32.dll" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/CodeCoverage/covrun32.dll". Beginning retry 1 in 1000ms. The process cannot access the file '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/CodeCoverage/covrun32.dll' because it is being used by another process.  [/build/source/tests/Games/NexusMods.Games.MountAndBlade2Bannerlord.Tests/NexusMods.Games.MountAndBlade2Bannerlord.Tests.csproj]
/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/Microsoft.CodeCoverage.targets(23,5): warning MSB3026: Could not copy "/build/tmp.dKYu2Rpy6p/.nuget/packages/microsoft.codecoverage/17.9.0/build/netstandard2.0/CodeCoverage/CodeCoverage.config" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/CodeCoverage/CodeCoverage.config". Beginning retry 1 in 1000ms. The process cannot access the file '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/CodeCoverage/CodeCoverage.config' because it is being used by another process.  [/build/source/tests/Games/NexusMods.Games.MountAndBlade2Bannerlord.Tests/NexusMods.Games.MountAndBlade2Bannerlord.Tests.csproj]
  NexusMods.Games.DarkestDungeon.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
/nix/store/4mmpxd5ifbin8dxi2pd8hrsgbycr097a-dotnet-sdk-8.0.301/sdk/8.0.301/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Publish.targets(351,5): error MSB3021: Unable to copy file "/build/source/src/ArchiveManagement/NexusMods.FileExtractor/runtimes/linux-x64/native/7zz" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/runtimes/linux-x64/native//7zz". Access to the path '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/runtimes/linux-x64/native/7zz' is denied. [/build/source/tests/Games/NexusMods.Games.Generic.Tests/NexusMods.Games.Generic.Tests.csproj]
  NexusMods.Games.Sifu.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.FOMOD.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.BladeAndSorcery.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Abstractions.Installers.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Abstractions.GuidedInstallers.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.DataModel -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.CrossPlatform.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.DarkestDungeon -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Abstractions.Games.Diagnostics -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Networking.Downloaders -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.StandardGameLocators.TestHelpers -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.DataModel.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
/nix/store/4mmpxd5ifbin8dxi2pd8hrsgbycr097a-dotnet-sdk-8.0.301/sdk/8.0.301/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Publish.targets(351,5): error MSB3021: Unable to copy file "/build/source/src/ArchiveManagement/NexusMods.FileExtractor/runtimes/linux-x64/native/7zz" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/runtimes/linux-x64/native//7zz". Access to the path '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/runtimes/linux-x64/native/7zz' is denied. [/build/source/tests/Networking/NexusMods.Networking.Downloaders.Tests/NexusMods.Networking.Downloaders.Tests.csproj]
  NexusMods.Networking.NexusWebApi.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.CLI.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.BethesdaGameStudios.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.TestFramework -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.RedEngine.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
/nix/store/4mmpxd5ifbin8dxi2pd8hrsgbycr097a-dotnet-sdk-8.0.301/sdk/8.0.301/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Publish.targets(351,5): error MSB3021: Unable to copy file "/build/source/src/ArchiveManagement/NexusMods.FileExtractor/runtimes/linux-x64/native/7zz" to "/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/runtimes/linux-x64/native//7zz". Access to the path '/nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/runtimes/linux-x64/native/7zz' is denied. [/build/source/tests/Games/NexusMods.Games.StardewValley.Tests/NexusMods.Games.StardewValley.Tests.csproj]
  NexusMods.Games.AdvancedInstaller.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.MountAndBlade2Bannerlord.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.AdvancedInstaller.UI.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.App.UI -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.AdvancedInstaller.UI -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Games.FOMOD.UI -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.Themes.NexusFluentDark -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.App -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  NexusMods.UI.Tests -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
  Examples -> /nix/store/q60chbfvsha3cf19zasj7aw4n4rd20hq-nexusmods-app-0.4.1/lib/nexusmods-app/
```

Mic92 commented:
Also fails in ofborg.

MattSturgeon commented:
> I get the following error:

Thanks for reviewing! There shouldn't be any functional changes to the package (or its tests) in this PR, other than a slight refactor to use `mapAttrs`.

I suspect the issue may already be in nixpkgs? If so, hopefully the update PR will resolve: #318647

Otherwise is there anything I should be checking on my end? (Happy to chat on matrix if easier)
